From f4308163447d26e6f9507844f21b06b7391d521f Mon Sep 17 00:00:00 2001
From: Jan Nieuwenhuizen <janneke@gnu.org>
Date: Fri, 6 Nov 2009 10:21:06 +0100
Subject: [PATCH] layout::Edit::GetSelection: only a noop if no XTextComponent.  Fixes n#552450.

   * Modified     toolkit/source/layout/vcl/wfield.cxx

Also fix SetSelection, both reversing the functionality.  This is
probably a post-layoutdialogs02-cws or layoutdialogs2-integration
fixup.
---
 toolkit/source/layout/vcl/wfield.cxx |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git toolkit/source/layout/vcl/wfield.cxx toolkit/source/layout/vcl/wfield.cxx
index 5187bad..442454c 100644
--- toolkit/source/layout/vcl/wfield.cxx
+++ toolkit/source/layout/vcl/wfield.cxx
@@ -109,7 +109,7 @@ Edit::~Edit ()
 void Edit::SetSelection( Selection const& rSelection )
 {
 #if LAYOUT_API_CALLS_HANDLER
-    if ( !getImpl().mxEdit.is() )
+    if ( getImpl().mxEdit.is() )
         getImpl().mxEdit->setSelection( awt::Selection( rSelection.Min(), rSelection.Max() ) );
 #else /* !LAYOUT_API_CALLS_HANDLER */
     GetEdit ()->SetSelection (rSelection);
@@ -129,7 +129,7 @@ void Edit::SetText( OUString const& rStr )
 
 String Edit::GetText() const
 {
-    if ( !getImpl().mxEdit.is() )
+    if ( getImpl().mxEdit.is() )
         return getImpl().mxEdit->getText();
     return OUString();
 }
-- 
1.6.3.3

